### PR TITLE
Fix various warnings & correctness issues

### LINKED ISSUE #11428: CMake: Improper set argument for VERSION_EXTRA
##### Minetest version
```
e9bc59e
```

##### OS / Hardware

Operating system: Linux Mint 19.3 Cinnamon
CPU: Intel Xeon X5680

GPU model: RTX 2060
OpenGL version: 4.6.0

##### Summary

CMakeLists.txt, line 22 should have parentheses so the value is properly set (otherwise it isn't applied). The corrected line is: `set(VERSION_STRING "${VERSION_STRING}-${VERSION_EXTRA}")`

### LINKED ISSUE #11751: Signed integer underflow in src/unittest/test_utilities.cpp
##### Minetest version
commit 80d12dbed

##### Summary
In the last loop iteration from 1 << 31 (INT_MIN) the value of 1 is substracted here:
```
        for (int exponent = 2; exponent <= 31; ++exponent) {                         
                UASSERT(is_power_of_two((1 << exponent) - 1) == false);
                UASSERT(is_power_of_two((1 << exponent)) == true);
                UASSERT(is_power_of_two((1 << exponent) + 1) == false);
        }
```

I believe this to be illegal.

I verified that 1 << 31 was actually INT_MIN with this program:
```
#include <iostream>
#include <climits>

int main() {
  int i = 1 << 31;
  std::cout << i << std::endl;
  std::cout << INT_MIN << std::endl;
  return 0;
}
```

It prints:
```
-2147483648
-2147483648
```

### LINKED ISSUE #11797: In src/client/game.cpp line 1386, “delete text” should be “delete[] text”
##### Minetest version
commit 1dc1305ada07da8c2a278b46a34d58af86184af9

##### Summary
Quote @sfan5:
> you'd look inside wgettext, find that it uses new[] and conclude that the counterpart must be delete[]

##### Steps to reproduce
Compile minetest with the compile option `-fsanitize=address` and run it.

##### Patch
```
rom d4ed792958594d0fde657304b3ce162061d00fd3 Mon Sep 17 00:00:00 2001
From: Nils Dagsson Moskopp <nils@dieweltistgarnichtso.net>
Date: Tue, 30 Nov 2021 19:57:13 +0100
Subject: [PATCH] Properly deallocate memory

---
 src/client/game.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/client/game.cpp b/src/client/game.cpp
index fb993d92f..7815a6195 100644
--- a/src/client/game.cpp
+++ b/src/client/game.cpp
@@ -1383,7 +1383,7 @@ bool Game::createClient(const GameStartData &start_data)
 		str += L" [";
 		str += text;
 		str += L"]";
-		delete text;
+		delete text[];
 	}
 	str += L" [";
 	str += driver->getName();
-- 
2.20.1

```

### LINKED ISSUE #11800: Unit test fails if language set to German
##### Minetest version
commit 413be76

##### Summary
```
======== Testing module TestGettext
Test assertion failed: buf == "Viewing range changed to 12"
    at test_gettext.cpp:30
    actual:   Sichtweite geändert auf 12
    expected: Viewing range changed to 12
[FAIL] testFmtgettext - 0ms
```

##### Steps to reproduce
Start minetest with run--unittests and `LANG` environment variable set to `de_DE.UTF-8` to see it fail.

Start with `LANG` set to `C` to see it succeed.

### LINKED ISSUE #11807: comparison of integer expressions of different signedness in src/client/content_cao.cpp
##### Minetest version
Commit 1dc1305ada07da8c2a278b46a34d58af86184af9

##### OS / Hardware
Operating system: Debian GNU/Linux
CPU: Intel(R) Core(TM) Duo CPU      T2400  @ 1.83GHz

##### Summary

During a build, I get the output:
```
src/client/content_cao.cpp: In member function ‘virtual void GenericCAO::addToScene(ITextureSource*, irr::scene::ISceneManager*)’:
src/client/content_cao.cpp:853:22: warning: comparison of integer expressions of different signedness: ‘irr::s32’ {aka ‘int’} and ‘irr::u32’ {aka ‘unsigned int’} [-Wsign-compare]
    for (s32 i = 1; i < mat_count; i++) {
```

I believe the `s32` should be `u32` here.

##### Steps to reproduce
1. Build commit 1dc1305ada07da8c2a278b46a34d58af86184af9.
2. Wait and see.

### LINKED ISSUE #11837: Emerge threads aren't started for two seconds for no reason
`server.cpp`:
```c++
	/*
		Trigger emergethread (it somehow gets to a non-triggered but
		bysy state sometimes)
	*/
	{
		float &counter = m_emergethread_trigger_timer;
		counter += dtime;
		if (counter >= 2.0) {
			counter = 0.0;

			m_emerge->startThreads();
		}
	}
```
After the first call, which takes 2s to happen, the event manager will set a flag to true; all subsequent calls are no-ops. The call should be done instantaneously, without a timer, and ideally outside of the game loop.

### PR DESCRIPTION
**Pull request by sfan5**
fixes #11428, fixes #11751, fixes #11800, fixes #11797, fixes #11807, fixes #11837

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,7 +26,7 @@ set(DEVELOPMENT_BUILD TRUE)
 
 set(VERSION_STRING "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
 if(VERSION_EXTRA)
-	set(VERSION_STRING ${VERSION_STRING}-${VERSION_EXTRA})
+	set(VERSION_STRING "${VERSION_STRING}-${VERSION_EXTRA}")
```
Comment (erlehmann) [inline CMakeLists.txt:29]: Please explain in the commit message what bug this change fixes. It is not obvious to me.

File: src/unittest/test_gettext.cpp [hunk 2]
```diff
@@ -7,13 +7,12 @@ class TestGettext : public TestBase
...
 	TestGettext() {
 		TestManager::registerTestModule(this);
-  }
+	}
 
 	const char *getName() { return "TestGettext"; }
 
 	void runTests(IGameDef *gamedef);
 
-	void testSnfmtgettext();
```
Comment (erlehmann) [inline src/unittest/test_gettext.cpp:16]: Why was this removed?

**Review by erlehmann — COMMENTED** [review]
Consider including a short explanation of non-obvious bugfixes in the corresponding commit message.

File: CMakeLists.txt (hunk 1 above)
Comment (sfan5) [reply CMakeLists.txt:29]: I don't think there's an actual bug here but it's better to quote the arguments.

File: src/unittest/test_gettext.cpp (hunk 2 above)
Comment (sfan5) [reply src/unittest/test_gettext.cpp:16]: Unused function definition

**Review by nerzhul — APPROVED** [review]
